sdk/python/lib/pulumi/runtime/stack.py

```diff
@@ -104,36 +96,23 @@ async def wait_for_rpcs(await_all_outstanding_tasks=True) -> None:
 
         # If the RPCs have successfully completed, now await all remaining outstanding tasks.
         if await_all_outstanding_tasks:
-            outstanding_tasks = _get_running_tasks()
-            if len(outstanding_tasks) == 0:
-                log.debug("No outstanding tasks to complete")
-            else:
+            while len(SETTINGS.outputs) > 0:
+                await asyncio.sleep(0)
```

Comment:
Out of curiosity, why is this necessary? Is this some pythonism I don't know about?

Reply:
It's so other things in the event loop can run. Just read it like "thread yield"